### PR TITLE
feat: [IOPID-2322] - CieID Error handling

### PR DESCRIPTION
**Pull request by ChrisMattew**
## Short description
Added error and loading handling

## List of changes proposed in this pull request
- Updated scripts `cie-ios:prod` and `cie-ios:dev`
- Replaced the default `react-native-webview` loader with the `LoadingOverlay` component in the `renderLoading` method
- Configured `onHttpError` and `onError` to use `handleLoadingError` function
- Added `isAuthenticationUrl` utility function to detect when to open the `CieID` app
- Added `handleLoginFailure` function to properly redirect to the `AuthErrorScreen`
- Added 4 dedicated errors in the `AuthErroComponent`

## Demo

<details><summary>iOS</summary>

### Operation canceled
|NFC Available|NFC not available|From Deep Link|
|--------------|-----------------|----------------|
|<video src="https://github.com/user-attachments/assets/95479823-c1f3-4533-8b9a-becbc1e59d1d" />|<video src="https://github.com/user-attachments/assets/b0e3f141-c450-40e3-b611-642a8687f629" />|<video src="https://github.com/user-attachments/assets/1326461f-4d1d-45f1-a088-6f9dfce02e6f" />|
### Invalid operation
|From Deep Link|
|----------------|
|<video src="https://github.com/user-attachments/assets/c9ab7906-6851-41bc-91e7-7b6461c503d3" />|
</details>

<details><summary>Android</summary>

### Operation canceled

|NFC Available|NFC not available|App2App|
|--------------|-----------------|--------------------|
|<video src="https://github.com/user-attachments/assets/744ad8cc-502c-4714-a5d6-41ad72aa9bd4" />|<video src="https://github.com/user-attachments/assets/39d61e8e-9e70-46b8-a8c2-97640e17ae98" />|<video src="https://github.com/user-attachments/assets/1575b45e-21ba-473c-af83-370eb8a5dbc7" />|

</details>

## How to test
Run the app in the local environment, and emulate what shown in the Demo videos

>[!Important]
>Currently, the following errors are expected:
>
>### Android
>- `CIEID_OPERATION_CANCEL`: Currently handled only on Android in the production environment.
>
>### iOS (Deep link)
>- Operation canceled
>- Operation invalid
>
>### Both platforms
>Clicking the "Cancel" CTA during the loading of the login screen.

### CODE REVIEW

**Comment by ChrisMattew** [comment]
@mariateresaventura copy review needed 🙏🏻

**Review by Ladirico — CHANGES_REQUESTED** [review]

File: ts/screens/authentication/components/AuthErrorComponent.tsx [hunk 1]
```diff
@@ -8,6 +8,13 @@ import UnlockAccessComponent, {
   UnlockAccessProps
 } from "../UnlockAccessComponent";
 
+// https://pagopa.atlassian.net/wiki/spaces/IAEI/pages/1305084137/Analisi+della+gestione+degli+errori+CieID+L2+e+L3#iOS
+export const IOS_OPERATION_CANCELED_MESSAGE =
```
[the remark this answers is not in the document]
Comment (ChrisMattew) [reply ts/screens/authentication/components/AuthErrorComponent.tsx:12]: I agree!

File: ts/features/cieLogin/components/CieIdLoginWebView.tsx [hunk 2]
```diff
@@ -47,14 +67,39 @@ const CieIdLoginWebView = ({ spidLevel, isUat }: CieIdLoginProps) => {
...
       screen: ROUTES.AUTHENTICATION_CIE_ID_ERROR
     });
   }, [navigation]);
 
   const { shouldBlockUrlNavigationWhileCheckingLollipop, webviewSource } =
     useLollipopLoginSource(navigateToCieIdAuthenticationError, loginUri);
 
+  const handleLoginFailure = useCallback(
+    (code?: string) => {
+      // TODO: Check missing SAML response (error message)
```
[the remark this answers is not in the document]
Comment (ChrisMattew) [reply ts/features/cieLogin/components/CieIdLoginWebView.tsx:80]: Good catch, thanks!

**Comment by mariateresaventura** [comment]
> @mariateresaventura copy review needed 🙏🏻

Done!

**Review by Ladirico — APPROVED** [review]
LGTM!

**Pull request merged**